### PR TITLE
Feature/attr apply

### PR DESCRIPTION
**Pull request by realloc**
Adding recipe to define ssl_certificate resources from node attributes.

### CODE REVIEW

**Comment by zuazo** [comment]
Thanks @realloc for your PR.

Could you fix the RuboCop offenses and add some tests (test-kitchen or ChefSpec)?

* [CONTRIBUTING documentation](https://github.com/onddo/ssl_certificate-cookbook/blob/master/CONTRIBUTING.md)
* [TESTING documentation](https://github.com/onddo/ssl_certificate-cookbook/blob/master/TESTING.md)
* [ChefSpec tests](https://github.com/onddo/ssl_certificate-cookbook/tree/master/spec/recipes)
* [test-kitchen tests](https://github.com/onddo/ssl_certificate-cookbook/tree/master/test)

**Comment by realloc** [comment]
Fixed rubocop. Added simple ChefSpec test.

File: recipes/attr_apply.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+       key_bag key_item key_item_key key_encrypted
+       key_secret_file key_content cert_path cert_name
+       cert_dir cert_source cert_bag cert_item cert_item_key
+       cert_encrypted cert_secret_file cert_content
+       subject_alternate_names chain_path chain_name chain_dir
+       chain_source chain_bag chain_item chain_item_key
+       chain_encrpted chain_secret_file chain_content
+       chain_combined_name chain_combined_path ca_cert_path ca_key_path
+    ).each do |attr|
+      send(attr, item_hash[attr])  if item_hash[attr]
```
Comment (zuazo) [inline recipes/attr_apply.rb]: If I understand this correctly, something like this should do the same:

```ruby
node['ssl_certificate']['items'].each do |item_hash|
  ssl_certificate item_hash[:name] do
    namespace item_hash
  end
end
```
Am I missing something?

See [the namespaces documentation](https://github.com/onddo/ssl_certificate-cookbook#namespaces).

**Comment by realloc** [comment]
Yes, your code does the same thing better.
I missed namespaces, was thinking about filtering attribute and allowing to set only pre-defined set of them.

**Comment by zuazo** [comment]
Released in [1.4.0](https://github.com/onddo/ssl_certificate-cookbook/releases/tag/1.4.0). Thanks for your contribution!

**Pull request merged**